src/backends/git-gateway/API.js

```diff
@@ -27,6 +27,12 @@ export default class API extends GithubAPI {
       });
   }
 
+  checkPrivateRepo(url) {
+    const cacheBuster = new Date().getTime();
+    return fetch(`${url}?ts=${cacheBuster}`)
+      .then(response => response.ok);
```

Comment:
It might be a good idea to check if we get a 403 (rate limited) error here, and assume that the repo is private in that case so that thumbnails don't break.

Reply:
That makes sense.